pkg/hive/hive.go

```diff
@@ -156,6 +159,16 @@ func (h *Hive) SetEnvPrefix(prefix string) {
 	h.envPrefix = prefix
 }
 
+// AddConfigOverrides appends config override functions of type
+// func(*cfg) where cfg implements Flagger and has an associated
+// config cell. The parsed configuration is passed by pointer to
+// the override function which can then mutate it.
+//
+// This method is only meant to be used in tests.
+func (h *Hive) AddConfigOverrides(configOverrides ...any) {
```

Comment:
and similarly `...func(Flagger)`

Reply:
Then one cannot pass in functions like `func(cfg *MyConfig)` but instead would need to declare them as `func(cfg Flagger)` and do type casts. Further I'm pretty sure this would also cause type erasure and reflect would just see them as `func(Flagger)` instead of the concrete types we want.

Reply:
If we want slightly nicer type signatures, then we could do:
```
package hive
func AddConfigOverride[Cfg Flagger](h *Hive, override func(*Cfg)) { ... }
```

Reply:
oh good point re type erasure - But yeah I think I do prefer the latter for API discoverability; personally I mostly rely on types to get a quick understanding of what a function/method does so this would help if it works... The downside of course is that you now have to write the type parameter. Still, yeah I do prefer it